### ISSUE TITLE
nestjs-core instrumentation breaks Metadata reflection against ExecutionContext

### ISSUE THREAD

**Issue by DanStory** [open-telemetry/opentelemetry-js-contrib#633]
### What version of OpenTelemetry are you using?
@opentelemetry/api: v1.0.2
@opentelemetry/node: v0.24.0
@opentelemetry/instrumentation-nestjs-core: main (unreleased)
@nestjs: v7.6.6

### What version of Node are you using?
node: v12.22.5

### What did you do?
```js
@Injectable()
export class MetaGuard implements CanActivate {
  async canActivate (context: ExecutionContext): Promise<boolean> {
    const reflector = new Reflector();
    const meta = reflector.getAll('meta', [context.getClass(), context.getHandler()]);
    return meta.every(m => m !== undefined);
  }
}

@Controller('metadata')
@SetMetadata('meta', 'controller metadata')
@UseGuards(MetaGuard)
export class MetadataController {
  @Get()
  @SetMetadata('meta', 'handler metadata')
  get(): void {
    return;
  }
}

nestApp.useGlobalGuards(nestApp.get(MetaGuard))
```

### What did you expect to see?
`reflector.getAll('meta', [context.getClass(), context.getHandler()]);` returns `["controller metadata", "handler metadata"]`

### What did you see instead?
`reflector.getAll('meta', [context.getClass(), context.getHandler()]);` returns `[undefined, undefined]`

**Comment by austonpramodh** [comment]
I see that metadata assigned with the decorator also gets messed up.

```ts
export const isPublicMetaDataKey = "isPublic";
export const Public = (): CustomDecorator<string> => SetMetadata(isPublicMetaDataKey, true);
```

if I have `Public()` on a handler, `this.reflector.get<boolean>(isPublicMetaDataKey, context.getHandler());` doesn't get the value.

**Comment by danshapir** [comment]
Still not working

**Comment by dyladan** [comment]
Fixed by #796 will be released today
